### PR TITLE
[FIX] Make load_confounds return None for confounds if no cleaning strategy is provided

### LINKED ISSUE #4570: [BUG] When using NiftiLabelsMasker with an empty confounds DataFrame, the masker raises an error
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [X] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

- Linux Pop!_OS 22.04 LTS

### Python version

- [ ] 3.12
- [ ] 3.11
- [X] 3.10
- [ ] 3.9
- [ ] 3.8

### nilearn version

nilearn 0.10.4

### Expected behavior

I have a GUI that dynamically collects cleaning arguments. If I have no cleaning strategy
```
cleaning_args = {'strategy': []}
confounds, _ = nilearn.interfaces.fmriprep.load_confounds(img, **cleaning_args)
```
I will get an empty confounds data frame. If I then apply e.g. detrending:
```
masker = NiftiLabelsMasker(labels_img=atlas, detrend=True)
time_series = masker.fit_transform(img, confounds=confounds)
```
I would still expect this to work as the detrending should not depend on the confounds (?). However the program crashes with `sklearn.utils._param_validation.InvalidParameterError: The 'n' parameter of gen_even_slices must be an int in the range [1, inf). Got 0 instead.` (see stack trace below).

My current workaround is
```
if confounds.empty:
    confounds = None
```
however I am unsure if this could introduce additional issues in certain edge cases. It might be more appropriate to handle this internally. Please let me know if a PR would be welcome.

### Current behavior & error messages

This is what I got:

```bash
Traceback (most recent call last):
  File "/home/mibur/dfc-multiverse/testing_stuff/test_labelsmasker.py", line 11, in <module>
    time_series = masker.fit_transform(img, confounds=confounds)
  File "/home/mibur/.local/lib/python3.10/site-packages/sklearn/utils/_set_output.py", line 273, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
  File "/home/mibur/anaconda3/envs/dfc-multiverse/lib/python3.10/site-packages/nilearn/maskers/nifti_labels_masker.py", line 620, in fit_transform
    return self.fit(imgs).transform(
  File "/home/mibur/.local/lib/python3.10/site-packages/sklearn/utils/_set_output.py", line 273, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
  File "/home/mibur/anaconda3/envs/dfc-multiverse/lib/python3.10/site-packages/nilearn/maskers/base_masker.py", line 268, in transform
    return self.transform_single_imgs(
  File "/home/mibur/anaconda3/envs/dfc-multiverse/lib/python3.10/site-packages/nilearn/maskers/nifti_labels_masker.py", line 724, in transform_single_imgs
    region_signals, (ids, masked_atlas) = self._cache(
  File "/home/mibur/.local/lib/python3.10/site-packages/joblib/memory.py", line 353, in __call__
    return self.func(*args, **kwargs)
  File "/home/mibur/anaconda3/envs/dfc-multiverse/lib/python3.10/site-packages/nilearn/maskers/base_masker.py", line 139, in _filter_and_extract
    region_signals = cache(
  File "/home/mibur/.local/lib/python3.10/site-packages/joblib/memory.py", line 353, in __call__
    return self.func(*args, **kwargs)
  File "/home/mibur/anaconda3/envs/dfc-multiverse/lib/python3.10/site-packages/nilearn/signal.py", line 756, in clean
    confounds = standardize_signal(
  File "/home/mibur/anaconda3/envs/dfc-multiverse/lib/python3.10/site-packages/nilearn/signal.py", line 70, in standardize_signal
    signals = _detrend(signals, inplace=False) if detrend else signals.copy()
  File "/home/mibur/anaconda3/envs/dfc-multiverse/lib/python3.10/site-packages/nilearn/signal.py", line 284, in _detrend
    for batch in gen_even_slices(signals.shape[1], n_batches):
  File "/home/mibur/.local/lib/python3.10/site-packages/sklearn/utils/_param_validation.py", line 203, in wrapper
    validate_parameter_constraints(
  File "/home/mibur/.local/lib/python3.10/site-packages/sklearn/utils/_param_validation.py", line 95, in validate_parameter_constraints
    raise InvalidParameterError(
sklearn.utils._param_validation.InvalidParameterError: The 'n' parameter of gen_even_slices must be an int in the range [1, inf). Got 0 instead.
```

### Steps and code to reproduce bug

```python
from nilearn.maskers import NiftiLabelsMasker
from nilearn.interfaces.fmriprep import load_confounds
from nilearn import datasets

atlas = datasets.fetch_atlas_aal()["maps"]
img = "<img.nii.gz>" # has to be an image in an fmriprep outputs folder because of the confounds file
cleaning_args = {'strategy': []}

confounds, _ = load_confounds(img, **cleaning_args)
masker = NiftiLabelsMasker(labels_img=atlas, detrend=True)
time_series = masker.fit_transform(img, confounds=confounds)
```

### PR DESCRIPTION
**Pull request by mibur1**
- Closes #4570 

**Changes proposed in this pull request:**

If the user passes an empty cleaning strategy to `nilearn.interfaces.fmriprep.load_confounds`
- Display warning 
- Return `None` for confounds

Adding the check at the end of the function is the simplest solution (imo) as the workflow still needs to run for `sample_mask_out`. I'm not sure if a test is necessary, please let me know if I should provide one.

### CODE REVIEW

**Review by Remi-Gau — CHANGES_REQUESTED** [review]
small changes to do

**Comment by Remi-Gau** [comment]
> a test is necessary, please let me know if I should provide one.

Yeah better had a test for this to avoid any regression of the behavior in the future.

**Review by Remi-Gau — APPROVED** [review]

**Comment by Remi-Gau** [comment]
Let's wait for CI to run but this should be good to go.

File: nilearn/interfaces/fmriprep/load_confounds.py [hunk 1]
```diff
@@ -356,6 +360,11 @@ def load_confounds(
     if flag_single:
         confounds_out = confounds_out[0]
         sample_mask_out = sample_mask_out[0]
+
+    # If no strategy was provided, return None for confounds
+    if len(strategy) == 0:
+        confounds_out = None
```
[the remark this answers is not in the document]
Comment (bthirion) [reply nilearn/interfaces/fmriprep/load_confounds.py:368]: OK, this was discussed in #4570

**Review by bthirion — APPROVED** [review]
LGTM, thx.

**Pull request merged**